src/string_with_vars.ml

```diff
@@ -179,9 +179,9 @@ module Mode = struct
 end
 
 module Partial = struct
-  type nonrec 'a t =
-    | Expanded of 'a
-    | Unexpanded of t
+  type nonrec ('expanded, 'unexpanded) t =
```

Comment:
Why this change? The original type looked simpler and looking at the use cases for `Partial.t`, it seems that `'unexpanded` is always `String_with_vars.t`

Reply:
It's because before I wanted the type of `Ordered_set_lang.Unexpanded.expand` t o be:

```ocaml
val expand
  :  t
  -> dir:Path.t
  -> f:(String_with_vars.t -> (Value.t list, Strign_with_vars.t) String_with_partial.Partial.t))
  -> (expanded, Partial.t) String_with_vars.Partial.t
```

where the point was that we could short circuit the `Osl.Partial.t` step if we managed to expand everything. I never managed to get this to work, and I'm not even sure it's possible to do it efficiently after a recent try. So I will undo this change.